fabric-armor-api-v1/src/main/java/net/fabricmc/fabric/api/armor/v1/ArmorMaterial.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.armor.v1;
+
+import java.util.function.Supplier;
+
+import net.minecraft.item.ItemStack;
+
+public interface ArmorMaterial {
+	/**
+	 * The value returned should be the prefix of the armor layer textures.
+	 */
+	String getName();
+
+	int getEnchantability();
+
+	Supplier<ItemStack> getRepairIngredient();
```

Comment:
make this a Predicate instead

Reply:
why?

Reply:
you see disallowing multiple ingredients

Reply:
how about `Set<ItemStack>`? easier to make and easier to use too

Reply:
at least allow a way for Predicate

Reply:
done